### ISSUE TITLE
$this->request->getSession()->read() returns null also when the default is set

### ISSUE THREAD

**Issue by ironbone** [cakephp/cakephp#15279]
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 4.1.7
* Platform and Target: XAMPP

### What you did
$locale = $this->request->session()->read('locale', 'en_GB');

### What happened
If there is no session active - I call the page for the first time - the value of $locale is set to NULL

### What you expected to happen
I expect the value of $locale to be in my case 'en_GB'

**Comment by ADmad** [comment]
`ServerRequest` class doesn't have a `session()` method, the method is `getSession()`.

I also checked `$locale = $this->request->session()->read('locale', 'en_GB');` with latest 4.x release and it does return the default value as expected if `locale` hasn't been set earlier.

**Comment by ironbone** [comment]
Sorry my mistake. But also with `$this->request->getSession()->read('locale', 'en_GB');` it does not work.
The problematical situation is not when the 'locale' in not set in the session. The problem appears when there is no session.

To see the problem, first delete the cookie of the page from the browser and then try it. In this situation and the following code
```
public function home() {
      
        $this->Authorization->skipAuthorization();
        $session = $this->request->getSession();
        
        $locale = $session->read('locale', 'en_GB');
        
        I18n::setLocale($locale);

    }
```

I get an error because $locale is NULL.

**Comment by ADmad** [comment]
`$_SESSION` is always set once the session is started, which `Session::read()` method ensures it is.

I suggest you check using plain PHP that your session handling isn't broken in some way.

**Comment by ironbone** [comment]
By why when I use (of course this is not well written, I add just session_start()  )
```
public function home() {
       session_start();
        $this->Authorization->skipAuthorization();
        $session = $this->request->getSession();
        
        $locale = $session->read('locale', 'en_GB');
        
        I18n::setLocale($locale);

    }

```
then, in the situation after deleting all cookies, there is no error. It looks that if I do not write to the session nothing and only try to read the session is not started.

**Comment by ironbone** [comment]
You are wrong. There is a bug.
So what I did:
I have installed a fresh application:
`composer create-project --prefer-dist cakephp/app:~4.0 SessionTest`
I edited only the 
`PagesController.php`
I added the followin 3 lines as the first lines in the display() funsction

```
$locale = $this->request->getSession()->read('locale', 'en_GB');
var_dump($locale);
exit();
```
Then I deleted all my cookies connected with the server. After staring the application the value is NULL.

Just to be sure that everything is correct here is my repo: https://github.com/ironbone/SessionTest

The application from this repo is installed here: http://test.lllc.info/
There you do need to delete ant cookies.
